File: lib/src/model/hms_log_settings.dart
import '../enum/hms_log_level.dart';

class HMSLogSettings {
  final double maxDirSizeInBytes;
  final bool isLogStorageEnabled;
  final HMSLogLevel level;

  HMSLogSettings(
      {required this.maxDirSizeInBytes,
      required this.isLogStorageEnabled,
      required this.level});

  Map<String, dynamic> toMap() {
    return {
      'max_dir_size_in_bytes': maxDirSizeInBytes,
      'log_storage_enabled': isLogStorageEnabled,
      'log_level': HMSLogLevelValue.getValueFromHMSLogLevel(level)
    };
  }
}

File: packages/hmssdk_flutter/android/src/main/kotlin/live/hms/hmssdk_flutter/HMSLogSettings.kt
package live.hms.hmssdk_flutter

import live.hms.video.media.settings.HMSLogSettings
import live.hms.video.utils.HMSLogger

class HMSLogSettings {

    companion object {

        fun setLogSettings(maxDirSizeInBytes: Double, logStorageEnabled: Boolean, logLevel: String): HMSLogSettings {
            return HMSLogSettings(maxDirSizeInBytes = maxDirSizeInBytes.toLong(), isLogStorageEnabled = logStorageEnabled, level = getLogLevel(logLevel))
        }

        private fun getLogLevel(logLevel: String?): HMSLogger.LogLevel {
            return when (logLevel) {
                "error" -> HMSLogger.LogLevel.ERROR
                "off" -> HMSLogger.LogLevel.OFF
                "verbose" -> HMSLogger.LogLevel.VERBOSE
                "warn" -> HMSLogger.LogLevel.WARN
                else -> HMSLogger.LogLevel.OFF
            }
        }
    }
}
